hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/CustomTypeFactory.java

```diff
@@ -201,13 +201,17 @@ public static EndpointQualifier createEndpointQualifier(int type, String identif
         return EndpointQualifier.resolve(protocolType, identifier);
     }
 
-    public static SqlColumnMetadata createSqlColumnMetadata(String name, int type) {
+    public static SqlColumnMetadata createSqlColumnMetadata(String name, int type, boolean isNullableExists, boolean nullability) {
         SqlColumnType sqlColumnType = SqlColumnType.getById(type);
 
         if (sqlColumnType == null) {
             throw new HazelcastException("Unexpected SQL column type = [" + type + "]");
         }
 
-        return new SqlColumnMetadata(name, sqlColumnType);
+        if (isNullableExists) {
+            return new SqlColumnMetadata(name, sqlColumnType, nullability);
+        }
+
+        return new SqlColumnMetadata(name, sqlColumnType, true);
```

Comment:
So, if we are using SQL with the 4.2 client against the 4.0 server, the response will not have the nullable parameter and the client will automatically set the column as nullable. Is it okay? Just writing this to make sure that we handle this case properly

Reply:
By default, columns are nullable. The column becomes non-nullable olny if NOT NULL modifier applied during table creation or if you are selecting an expression, like in test above.

Reply:
Thanks for the explanation, we are good then